libyul/AsmParser.cpp

```diff
@@ -208,13 +236,12 @@ std::optional<std::pair<std::string_view, SourceLocation>> Parser::parseSrcComme
 		return std::nullopt;
 	}
 
-	solAssert(match.size() == 5, "");
-	std::string_view tail = _arguments.substr(static_cast<size_t>(match.position() + match.length()));
+	if (scanner.currentToken() == Token::StringLiteral || (scanner.currentToken() == Token::Illegal && scanner.currentError() == ScannerError::IllegalStringEndQuote))
+		tail = _arguments.substr(static_cast<size_t>(scanner.currentLocation().end));
+	else
+		tail = _arguments.substr(static_cast<size_t>(scanner.currentLocation().start));
 
-	if (match[4].matched && (
-		!boost::algorithm::ends_with(match[4].str(), "\"") ||
-		boost::algorithm::ends_with(match[4].str(), "\\\"")
-	))
+	if (scanner.currentToken() == Token::Illegal && scanner.currentError() == ScannerError::IllegalStringEndQuote)
```

Comment:
Are errors other than `IllegalStringEndQuote` possible here? If not, I think it would be better to assert that to make that clearer (and then you'd only need a check for illegal token). If they are, then perhaps they should be handed explicitly (or at least acknowledged in a comment).

Reply:
Yeah they're possible if I have something like `@src 0:111:222 /** something something`; this would pop up as an unterminated comment. But at least in this context here it'll be ignored and put towards the tail s.t. the outer loop over tail ends takes care of it and extracts potential subsequent tags

Reply:
Ah, ok then. Good to have the comment though, it wasn't clear to me that this was the case.